Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/EngineView.cpp

```diff
@@ -2,173 +2,452 @@
+		const auto properties = point.Properties();
+		const auto deviceType = point.PointerDevice().PointerDeviceType();
+		const auto position = point.Position();
+		const uint32_t x = position.X < 0 ? 0 : static_cast<uint32_t>(position.X);
+		const uint32_t y = position.Y < 0 ? 0 : static_cast<uint32_t>(position.Y);
+
+		if (point.PointerDevice().PointerDeviceType() == PointerDeviceType::Mouse)
+		{
+			if (!properties.IsLeftButtonPressed() &&
+				_pressedMouseButtons.find(Babylon::LEFT_MOUSE_BUTTON_ID) != _pressedMouseButtons.end())
+			{
+				_pressedMouseButtons.erase(Babylon::LEFT_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::LEFT_MOUSE_BUTTON_ID, false, x, y);
+			}
+
+			if (!properties.IsMiddleButtonPressed() &&
+				_pressedMouseButtons.find(Babylon::MIDDLE_MOUSE_BUTTON_ID) != _pressedMouseButtons.end())
+			{
+				_pressedMouseButtons.erase(Babylon::MIDDLE_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::MIDDLE_MOUSE_BUTTON_ID, false, x, y);
+			}
+
+			if (!properties.IsRightButtonPressed() &&
+				_pressedMouseButtons.find(Babylon::RIGHT_MOUSE_BUTTON_ID) != _pressedMouseButtons.end())
+			{
+				_pressedMouseButtons.erase(Babylon::RIGHT_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::RIGHT_MOUSE_BUTTON_ID, false, x, y);
+			}
+		}
+		else
+		{
+			const auto pointerId = point.PointerId();
+			Babylon::SetTouchButtonState(pointerId, false, x, y);
+		}
+	}
+
+	void EngineView::OnCompositionScaleChanged(SwapChainPanel const& sender, IInspectable const& /*object*/)
+	{
+		if (_compositionScaleX != sender.CompositionScaleX() || _compositionScaleY != sender.CompositionScaleY())
+		{
+			critical_section::scoped_lock lock(s_criticalSection);
+
+			// Store values so they can be accessed from a background thread.
+			_compositionScaleX = sender.CompositionScaleX();
+			_compositionScaleY = sender.CompositionScaleY();
+
+			// Recreate size-dependent resources when the composition scale changes.
+			CreateSizeDependentResources();
+
+			auto swapChainPanel = static_cast<SwapChainPanel>(*this);
+			auto swapChainPanelNative = swapChainPanel.as<ISwapChainPanelNative>().get();;
+			swapChainPanelNative->SetSwapChain(_swapChain.Get());
+		}
+	}
+
+	void EngineView::OnSuspending()
+	{
+		critical_section::scoped_lock lock(s_criticalSection);
+
+		::Microsoft::WRL::ComPtr<IDXGIDevice3> dxgiDevice;
+		s_d3dDevice->QueryInterface(__uuidof(IDXGIDevice3), &dxgiDevice);
+
+		// Hints to the driver that the app is entering an idle state and that its memory can be used temporarily for other apps.
+		dxgiDevice->Trim();
+	}
+
+	void EngineView::OnResuming()
+	{}
+
+	void EngineView::OnRendering()
+	{
+		Concurrency::critical_section::scoped_lock lock(s_criticalSection);
+
+		if (!_backBufferPtr)
+			return;
+
+		// Clear the back buffer and depth stencil view.
+		float clearColor[4]{ 0.392156899f, 0.584313750f, 0.929411829f, 1.000000000f };
+		s_d3dContext->ClearRenderTargetView(_backBufferPtr.Get(), clearColor);

```

Comment:
is this the blue background that can be seen when initializing? Would we prefer to have this as black?